uber/models.py

```diff
@@ -1215,7 +1222,8 @@ def is_transferable(self):
     def gets_free_shirt(self):
         return self.is_dept_head \
             or self.badge_type == c.STAFF_BADGE \
-            or self.staffing and (self.assigned_depts and not self.takes_shifts or self.weighted_hours >= 6)
+            or self.staffing and (self.assigned_depts and not self.takes_shifts or self.weighted_hours >= 6) \
+            or self.donation_tier in c.SHIRT_TIERS
```

Comment:
Whoops, this will double-count people who paid for shirts. I'll have to make sure to also check for amount_extra < SHIRT_LEVEL